nrf_modem/doc/ug_nrf_modem_porting_os.rst

```diff
@@ -169,13 +169,13 @@ This function is called by the Modem library to put a thread to sleep unconditio
 nrf_modem_os_timedwait()
 ------------------------
 
-This function is called by the Modem library when a timed context or sleep is required.
-A blind return value of 0 will make all the Modem library operations always blocking.
+This function is called by the Modem library to put a thread to sleep for a certain amount of time or until an event occurs.
 
-*Required actions* to be taken that do not block the operations:
+*Required actions*:
 
-* Start counting the time (this can be based on a Timer or Thread for instance).
-* In the case of a call to :c:func:`nrf_modem_os_event_notify` function, all waiting threads must wake up, reporting back the remaining time of the timer through the ``timeout`` variable.
+* Put the thread to sleep, if applicable, or otherwise wait until the amount time specified by ``timeout`` has elapsed, or until :c:func:`nrf_modem_os_event_notify` is called with the same ``context`` parameter, or with ``context`` equal to ``0``.
```

Comment:
```suggestion
* Put the thread to sleep, if applicable, or otherwise wait until the amount of time specified by ``timeout`` has elapsed, or until :c:func:`nrf_modem_os_event_notify` is called with the same ``context`` parameter, or with ``context`` equal to ``0``
```